eventuate-cdc-data-producer-wrappers/src/main/java/io/eventuate/cdc/producer/wrappers/kafka/TopicPartitionMessage.java

```diff
@@ -0,0 +1,46 @@
+package io.eventuate.cdc.producer.wrappers.kafka;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+public class TopicPartitionMessage {
+  private String topic;
+  private String key;
+  private String body;
+
+  private CompletableFuture<Object> future = new CompletableFuture<>();
+
+  public TopicPartitionMessage(String topic, String key, String body) {
+    this.topic = topic;
+    this.key = key;
+    this.body = body;
+  }
+
+  public String getTopic() {
+    return topic;
+  }
+
+  public String getKey() {
+    return key;
+  }
+
+  public String getBody() {
+    return body;
+  }
+
+  public CompletableFuture<Object> getFuture() {
+    return future;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    return EqualsBuilder.reflectionEquals(this, o);
```

Comment:
Do these equals()/hashCode() methods make sense?
It's very strange to equals on a completableFuture

Reply:
I forgot to exclude future from the equals, thank you.
It is needed for retry on fail.
If publisher sends the same message that last failed, wrapper will resend it.

Reply:
I am not sure what retrying to be handled at this higher-level.
The low-level sending logic can retry (or just the kafka publisher).

Reply:
Ok, I will remove it for now